engine/src/test/java/io/camunda/zeebe/engine/processing/incident/OutputMappingIncidentTest.java

```diff
@@ -0,0 +1,203 @@
+    // given
+    ENGINE.deployment().withXmlResource(bpmnModelInstance).deploy();
+
+    // when
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    if (createsJob) {
+      ENGINE.job().withType("type").ofInstance(processInstanceKey).complete();
+    }
+
+    // then
+    final var failureCommand =
+        RecordingExporter.processInstanceRecords()
+            .withElementId(elementId)
+            .withIntent(ProcessInstanceIntent.COMPLETE_ELEMENT)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+
+    final var incidentEvent =
+        RecordingExporter.incidentRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(IncidentIntent.CREATED)
+            .getFirst();
+
+    assertThat(incidentEvent.getKey()).isGreaterThan(0);
+    assertThat(incidentEvent.getSourceRecordPosition()).isEqualTo(failureCommand.getPosition());
+
+    Assertions.assertThat(incidentEvent.getValue())
+        .hasErrorType(ErrorType.IO_MAPPING_ERROR)
+        .hasBpmnProcessId(PROCESS_ID)
+        .hasProcessInstanceKey(processInstanceKey)
+        .hasElementId(elementId)
+        .hasElementInstanceKey(failureCommand.getKey())
+        .hasVariableScopeKey(failureCommand.getKey());
+
+    assertThat(incidentEvent.getValue().getErrorMessage())
+        .contains("no variable found for name 'foo'");
+  }
+
+  @Test
+  public void shouldResolveIncidentForOutputMappingFailure() {
+    // given
+    ENGINE.deployment().withXmlResource(bpmnModelInstance).deploy();
+
+    final long processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    if (createsJob) {
+      ENGINE.job().withType("type").ofInstance(processInstanceKey).complete();
+    }
+
+    final var incidentRecord =
+        RecordingExporter.incidentRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(CREATED)
+            .getFirst();
+
+    // when
+    ENGINE.variables().ofScope(processInstanceKey).withDocument(Maps.of(entry("foo", 1))).update();
+    final Record<IncidentRecordValue> resolvedIncidentRecord =
+        ENGINE.incident().ofInstance(processInstanceKey).withKey(incidentRecord.getKey()).resolve();
+
+    // then
+    assertThat(resolvedIncidentRecord.getKey()).isEqualTo(incidentRecord.getKey());
+    Assertions.assertThat(resolvedIncidentRecord.getValue())
+        .hasErrorType(ErrorType.IO_MAPPING_ERROR)
+        .hasBpmnProcessId(PROCESS_ID)
+        .hasProcessInstanceKey(processInstanceKey)
+        .hasElementId(elementId);
+    assertThat(resolvedIncidentRecord.getValue().getErrorMessage())
+        .contains("no variable found for name 'foo'");
+    assertTrue(
+        RecordingExporter.processInstanceRecords()
+            .withProcessInstanceKey(processInstanceKey)
+            .withIntent(ProcessInstanceIntent.ELEMENT_COMPLETED)
+            .exists());
```

Comment:
:x: We need to change the query to verify that the process instance is completed. 

Currently, it queries all process instance records of the given process instance. It will find the start event first and return `true`. We need to restrict the query either by the element type or by `filterRootScope()`.

For example:

```
      assertTrue(
        RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_COMPLETED)
            .withElementType(BpmnElementType.PROCESS)
            .withProcessInstanceKey(processInstanceKey)
            .exists());
```